src/main/java/org/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -41,4 +42,13 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Book findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.get(Book.class, id);
```

Comment:
1. you should return Optional here
2. you can leverage `fromSession` method in SessionFactory

will be like:
```
return sessionFactory.fromSession(s -> Optional.ofNullable(s.get(Book.class, id)));
```